server/src/test/java/org/elasticsearch/snapshots/SnapshotShardsServiceIT.java

```diff
@@ -0,0 +1,119 @@
+
+        final int shards = between(1, 10);
+        execute("create table doc.test(x integer) clustered into ? shards with (number_of_replicas=0)", new Object[]{shards});
+
+        ensureGreen();
+        final int numDocs = scaledRandomIntBetween(50, 100);
+        for (int i = 0; i < numDocs; i++) {
+            execute("insert into doc.test values(?)", new Object[]{i});
+        }
+
+        logger.info("--> blocking repository");
+        String blockedNode = blockNodeWithIndex("repo", "test");
+        dataNodeClient().admin().cluster().prepareCreateSnapshot("repo", "snapshot")
+            .setWaitForCompletion(false)
+            .setIndices("test")
+            .get();
+        waitForBlock(blockedNode, "repo", TimeValue.timeValueSeconds(60));
+
+        final SnapshotId snapshotId = client().admin().cluster().prepareGetSnapshots("repo").setSnapshots("snapshot")
+            .get().getSnapshots().get(0).snapshotId();
+
+        logger.info("--> start disrupting cluster");
+        final NetworkDisruption networkDisruption = new NetworkDisruption(new NetworkDisruption.TwoPartitions(masterNode, dataNode),
+                                                                          NetworkDisruption.NetworkDelay.random(random()));
+        internalCluster().setDisruptionScheme(networkDisruption);
+        networkDisruption.startDisrupting();
+
+        logger.info("--> unblocking repository");
+        unblockNode("repo", blockedNode);
+
+        // Retrieve snapshot status from the data node.
+        SnapshotShardsService snapshotShardsService = internalCluster().getInstance(SnapshotShardsService.class, blockedNode);
+        assertBusy(() -> {
+            final Snapshot snapshot = new Snapshot("repo", snapshotId);
+            List<IndexShardSnapshotStatus.Stage> stages = snapshotShardsService.currentSnapshotShards(snapshot)
+                .values().stream().map(status -> status.asCopy().getStage()).collect(Collectors.toList());
+            assertThat(stages, hasSize(shards));
+            assertThat(stages, everyItem(equalTo(IndexShardSnapshotStatus.Stage.DONE)));
+        }, 30L, TimeUnit.SECONDS);
+
+        logger.info("--> stop disrupting cluster");
+        networkDisruption.stopDisrupting();
+        internalCluster().clearDisruptionScheme(true);
+
+        assertBusy(() -> {
+            GetSnapshotsResponse snapshotsStatusResponse = client().admin().cluster()
+                .prepareGetSnapshots("repo")
+                .setSnapshots("snapshot").get();
+            SnapshotInfo snapshotInfo = snapshotsStatusResponse.getSnapshots().get(0);
+            logger.info("Snapshot status [{}], successfulShards [{}]", snapshotInfo.state(), snapshotInfo.successfulShards());
+            assertThat(snapshotInfo.state(), equalTo(SnapshotState.SUCCESS));
+            assertThat(snapshotInfo.successfulShards(), equalTo(shards));
```

Comment:
Would it be possible to use the `failures` column of the `sys.snapshots` table for these kind of assertions?